chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -572,9 +595,13 @@ public void joinChannel(String channelName) {
      */
     public void leaveChannel(String channelName) {
         String lowerChannelName = channelName.toLowerCase();
-        if (channelCache.remove(lowerChannelName)) {
+        if (currentChannels.remove(lowerChannelName)) {
             sendCommand("part", "#" + lowerChannelName);
             log.debug("Leaving Channel [{}].", lowerChannelName);
+
+            // clear cache
+            channelIdToChannelName.remove(channelNameToChannelId.get(lowerChannelName));
```

Comment:
If we do `PART` before `ROOMSTATE` occurs, name => id mapping will not exist so this will throw an NPE